### PR TITLE
Changed host from 'HTTP_HOST' to 'SERVER_NAME'

### PR DESCRIPTION
**Pull request by jakeasmith**
I ran into an issue where Url::get() was including the port twice `http://example.com:1180:1180/`. The value of `$_SERVER['HTTP_HOST']` is taken directly from the `Host: HTTP` request header and it appears some browsers (Chrome, in my case) include the port. `$_SERVER['SERVER_NAME']` gets its value from the virtual host config without the port.

Let me know if there is a better way of fixing this and I will amend the PR.

### CODE REVIEW

**Comment by harikt** [comment]
I am also looking here http://stackoverflow.com/questions/2297403/http-host-vs-server-name

**Comment by pmjones** [comment]
I can see where something like this would be good to have.  Some additional considerations:
1. Use `SERVER_NAME` only when `HTTP_HOST` is not present.
2. When the result of `SERVER_NAME || HTTP_HOST` ends in a colon and digits, strip them off and retain for helping to determine the port number.

Thoughts?

**Comment by jakeasmith** [comment]
Good feedback. I'll update the PR this weekend.

**Comment by jakeasmith** [comment]
Okay, I added some tests around this to reveal the problem and then took a stab at implementing the fix based on your comments. Take a look, let me know what you think.

**Comment by harikt** [comment]
Thanks. This looks good to me. We will hear from Paul.

One a side note : Are you running the cs-fixer ? If so for the next time, don't do. It will make hard to look the diff with white space.

Thank you.

File: tests/src/Request/UrlTest.php [hunk 1]
```diff
@@ -7,36 +7,81 @@ protected function newUrl($server = array())
...
+                    'REQUEST_URI' => '/foo?bar=baz'
+                )
+            ),
+            array(
+                'http://example.com:1180/foo?bar=baz',
+                array(
+                    'SERVER_NAME' => 'example.com',
+                    'SERVER_PORT' => '1180',
+                    'REQUEST_URI' => '/foo?bar=baz'
+                )
```
Comment (harikt) [inline tests/src/Request/UrlTest.php:83]: one more test to add can be an IP address .

**Comment by jakeasmith** [comment]
Hm. I hadn't noticed the whitespace changes. Must be my IDE settings. You can [ignore whitespace in the github diff](https://github.com/auraphp/Aura.Web/pull/26/files?w=1) by adding ?w=1 to the URL.

**Comment by harikt** [comment]
That is a great suggestion.

Thank you @jakeasmith .

**Comment by harikt** [comment]
Wondering the whitespace is not shown again now with ? w=0. 
Is it a settings applied to the user ?

**Pull request merged**